mods/mc_worldmanager/realm/realm.lua

```diff
@@ -51,67 +51,156 @@ function Realm:New(name, size, height)
 
     Realm.realmCount = this.ID
 
-    -- Calculate where on the realm grid we are located; based on our realm ID
-    local realmLocation = { x = 0, z = 0 }
-    realmLocation.x = this.ID % 10
-    realmLocation.z = math.ceil(this.ID / 10)
+    local gridStartPos, gridEndPos = Realm.CalculateStartEndPosition(area)
 
     -- Calculate our world position based on our location on the realm grid
-    this.StartPos.x = -20000 + (realmSize * realmLocation.x) + (realmBuffer * realmLocation.x)
-    this.StartPos.z = -20000 + (realmSize * realmLocation.z) + (realmBuffer * realmLocation.z)
+    this.StartPos = Realm.gridToWorldSpace(gridStartPos)
+    this.EndPos = Realm.gridToWorldSpace(gridEndPos)
 
-    -- Ensures that a realm size is no larger than our maximum size.
-    local finalRealmSize = math.min(realmSize, size)
-    local finalRealmHeight = math.min(realmHeight, height)
-
-    this.EndPos = { x = this.StartPos.x + finalRealmSize,
-                    y = this.StartPos.y + finalRealmHeight,
-                    z = this.StartPos.z + finalRealmSize }
 
     -- Temporary spawn point calculation
     this.SpawnPoint = { x = (this.StartPos.x + this.EndPos.x) / 2,
-                        y = ((this.StartPos.y + this.EndPos.y) / 2) + 2,
+                        y = (this.StartPos.y + 2),
                         z = (this.StartPos.z + this.EndPos.z) / 2 }
 
     setmetatable(this, self)
-    Realm.realmDict[self.ID] = this
+    Realm.realmDict[this.ID] = this
+
     Realm.SaveDataToStorage()
 
     return this
 end
 
+-- Online bin packing... A pretty challenging problem to solve.
```

Comment:
This will be a difficult one to break with a traceback to the binpacking algorithm. We would need to use and delete a lot of realms in the coordinate space.